caf-audit/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditConnection.java

```diff
@@ -30,7 +30,29 @@
 
     public ElasticAuditConnection(ConfigurationSource configSource) throws ConfigurationException {
         //  Get Elasticsearch configuration.
-        final ElasticAuditConfiguration config = configSource.getConfiguration(ElasticAuditConfiguration.class);
+        final ElasticAuditConfiguration config;
+        if (configSource == null) {
+            config = new ElasticAuditConfiguration();
+
+            // Get the Elasticsearch host and port values from env var
+            config.setHostAndPortValues(System.getProperty(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_AND_PORT_VALUES, System.getenv(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_AND_PORT_VALUES)));
+
+            // Get the Elasticsearch cluster name from env var else default to "elasticsearch-cluster"
+            String clusterName = System.getProperty(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_CLUSTER_NAME, System.getenv(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_CLUSTER_NAME));
```

Comment:
In this case I think the defaults do have to be in the code since this is a library and there may not be a configuration file.  In fact, in general there is no issue with defaults being in the code, and when there are sensible code defaults then the config files should default to 'undefined' to allow the code default to shine through.